### PR TITLE
Expand on evolutes in Section 8

### PR DESCRIPTION
**Pull request by armansito**
- Added a new stroke renderings figure demonstrating strong and weak correctness and outlines.
- Added a new Algorithm block with the outline of the WGSL `flatten_euler` function.
- Added a couple paragraphs that summarizes the logic.

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -449,14 +455,83 @@ \subsection{Pipeline Design} \label{subsec:pipeline-design}
...
 
-The joins between adjacent segments need to form a continuous outline when combined with the segments. This requires that a thread have access to the tangent vector at the end of its assigned segment \emph{and} the start tangent of the next adjacent segment. We require that the input segments within a subpath are stored in order which simplifies the access to the adjacent segment down to a buffer read at the next array offset. While we take care to output the \emph{outer} join segment (where the offset curves meet at a reflex angle) according to the join style, we output the inner join simply as a line segment.
+This procedure is sufficient to generate weakly correct stroke outlines. If the outline contains a cusp -- which can be determined by evaluating the curvature variation along the Euler spiral segment -- we output an evolute patch (as shown in Figure~\ref{fig:evolutes}) in order to achieve strong correctness.
```
Comment (armansito) [inline paper.tex]: > -- which can be determined by evaluating the curvature variation along the Euler spiral segment --

There might be a better way to summarize the computation of `t` (the evolute anchor point / cusp) via the `cusp0 * cusp1` multiplication and the sign comparison.  I think the new algorithm block demonstrates it clearly enough but maybe this should explain the state machine too.

File: paper.tex [hunk 2]
```diff
@@ -449,14 +455,83 @@ \subsection{Pipeline Design} \label{subsec:pipeline-design}
...
+            t0\_u $\gets$ t0\_u * 2u\;
+            dt $\gets$ dt * 0.5\;
+        }
+    }
+}
+\caption{Outline of the iterative cubic Bézier flattening routine invoked in our compute shader. A working WGSL implementation is provided in the supplemental materials.}
+\label{alg:flatten-cubic}
+\end{algorithm}
+
+All input segments first get converted to a cubic Bézier by degree elevation. Our flattening routine (see Algorithm~\ref{alg:flatten-cubic}) iteratively subdivides the input cubic segment in half, and at each step, computes an error estimate for geometric Hermite interpolation to an Euler spiral segment. Once the error satisfies a desired tolerance threshold, the Euler spiral segment is directly flattened to a polyline (or, alternately, converted to a sequence of arc segments) by invoking a subroutine that is parameterized by the stroke halfwidth (see \texttt{es\_seg\_flatten\_offset} in Algoritm~\ref{alg:flatten-cubic}). For a regular fill, this parameter is set to $0$. For strokes, the sign of the parameter determines the position of the parallel curve relative to the source segment as well as the winding direction of the output lines.
```
Comment (raphlinus) [inline paper.tex]: "iteratively subdivides...in half" suggests it always does at least one such subdivision. I'd add "when the error tolerance is exceeded" or "whenever".

File: paper.tex [hunk 3]
```diff
@@ -375,10 +377,10 @@ \subsection{Cusp handling}
 \end{figure}
 
 \begin{figure*}
-    \includegraphics[scale=0.2]{tricky_strokes}
-    \caption{Stroked cubic Béziers with cusps and near-cusps, rendered using our algorithm. The individual curves were adapted from the Skia test corpus.}
-    \Description{Stroked cubic Béziers with cusps and near-cusps, rendered using our algorithm.}
-    \label{fig:tricky_strokes}
+    \includegraphics[scale=0.19]{stroke_stills}
+    \caption{Renderings of stroked cubic Béziers exhibiting strong and weak correctness. The outlines generated by our GPU stroking compute shader are visualized in green and blue. Segments with a winding direction that points \emph{up} are colored blue while those pointing \emph{down} are colored green. The curves were adapted to our GPU input format from the \emph{stills} dataset by \citet{Nehab2020} and the Skia test corpus.}
```
Comment (raphlinus) [inline paper.tex]: I think "to our GPU input format" can be omitted.

**Review by raphlinus — APPROVED** [review]
Looks good modulo a couple small comments. I'll rebase my work on top (shouldn't conflict).

File: paper.tex (hunk 2 above)
Comment (armansito) [reply paper.tex]: PTAL at the new phrasing.

**Review by raphlinus — APPROVED** [review]
Yup, looks good, thanks! That's not an easy concept to write about.

**Pull request merged**